bentoml/_internal/frameworks/mlflow.py

```diff
@@ -205,19 +206,20 @@ def import_model(
             # For MLflow < 1.25
             from mlflow.tracking.artifact_utils import _download_artifact_from_uri
 
-            local_path = _download_artifact_from_uri(
+            local_path: str = _download_artifact_from_uri(
                 artifact_uri=model_uri, output_path=download_dir
             )
-
-        mlflow_model_path = bento_model.path_of(MLFLOW_MODEL_FOLDER)
-        # Rename model folder from original artifact name to fixed "mlflow_model"
-        shutil.move(local_path, mlflow_model_path)
-        # If the temp dir we created still exists now, we never needed it because
-        # the provided mlflow url must have provided enough path information.  Just
-        # delete it.  If it's not here, it means we needed it but it's been renamed
-        # by now so we don't need to remove it.
-        if os.path.isdir(download_dir):
+        finally:
+            if TYPE_CHECKING:
+                # We need to trick type checker here, since local_path
+                # will always be set
+                local_path = ""
```

Comment:
I think we have gone too far to please the type checker. 🙂 Maybe put this before the `try`?

Reply:
😄

Reply:
I think this is a true positive; it's possible if there are exceptions raised that `local_path` is not set.